static/app/components/events/eventReplay/replayClipPreview.tsx

```diff
@@ -0,0 +1,333 @@
+  // it. https://caniuse.com/fullscreen
+  const showFullscreenButton = screenfull.isEnabled;
+
+  const elem = useRef<HTMLDivElement>(null);
+  const mouseTrackingProps = useScrubberMouseTracking({elem});
+
+  const fullReplayUrl = {
+    pathname: normalizeUrl(`/organizations/${organization.slug}/replays/${replayId}/`),
+    query: {
+      referrer: getRouteStringFromRoutes(routes),
+      t_main: TabKey.ERRORS,
+      t: currentTime / 1000,
+    },
+  };
+
+  return (
+    <Fragment>
+      <StaticPanel>
+        <ReplayPlayer />
+      </StaticPanel>
+      <ErrorBoundary mini>
+        <ButtonGrid>
+          <ReplayPlayPauseBar />
+          <Container>
+            <TimeAndScrubberGrid>
+              <Time style={{gridArea: 'currentTime'}}>
+                {formatTime(currentTime - startTime)}
+              </Time>
+              <div style={{gridArea: 'timeline'}}>
+                <ReplayTimeline />
+              </div>
+              <StyledScrubber
+                style={{gridArea: 'scrubber'}}
+                ref={elem}
+                {...mouseTrackingProps}
+              >
+                <PlayerScrubber showZoomIndicators />
+              </StyledScrubber>
+              <Time style={{gridArea: 'duration'}}>
+                {duration ? formatTime(duration) : '--:--'}
+              </Time>
+            </TimeAndScrubberGrid>
+          </Container>
+          <ButtonBar gap={1}>
+            <LinkButton size="sm" to={fullReplayUrl}>
+              {t('See Full Replay')}
+            </LinkButton>
+            {showFullscreenButton ? (
+              <Button
+                size="sm"
+                title={isFullscreen ? t('Exit full screen') : t('Enter full screen')}
+                aria-label={isFullscreen ? t('Exit full screen') : t('Enter full screen')}
+                icon={
+                  isFullscreen ? <IconContract size="sm" /> : <IconExpand size="sm" />
+                }
+                onClick={toggleFullscreen}
+                {...fullReplayButtonProps}
+              />
+            ) : null}
+          </ButtonBar>
+        </ButtonGrid>
+      </ErrorBoundary>
+    </Fragment>
+  );
+}
+
+function ReplayClipPreview({
+  eventTimestampMs,
+  orgSlug,
+  replaySlug,
+  fullReplayButtonProps,
+}: Props) {
+  const {fetching, replay, replayRecord, fetchError, replayId} = useReplayReader({
+    orgSlug,
+    replaySlug,
+  });
+  const fullscreenRef = useRef(null);
+  const {toggle: toggleFullscreen} = useFullscreen({
+    elementRef: fullscreenRef,
+  });
+
+  const startTimestampMs = replayRecord?.started_at?.getTime() ?? 0;
+  const eventTimeOffsetMs = Math.abs(eventTimestampMs - startTimestampMs);
+
+  useRouteAnalyticsParams({
+    event_replay_status: getReplayAnalyticsStatus({fetchError, replayRecord}),
+  });
+
+  const clipWindow = useMemo(
+    () => ({
+      startTimeOffsetMs: Math.max(eventTimeOffsetMs - 10e3, 0),
+      durationMs: 15e3,
```

Comment:
can we use like the `10_000` `15_000` format or something? scientific `e3` is my least favorite by far far far. It really hides the constant which should not be a goal

Reply:
Also, i think we should be clamping the duration value so it doesn't run past the end of the replay.